src/number_theory/bernoulli.lean

```diff
@@ -328,3 +329,65 @@ begin
   rw [cast_mul, cast_mul, mul_div_mul_right _ _ hj, add_choose, cast_dvd_char_zero],
   exact factorial_mul_factorial_dvd_factorial_add i j,
 end
+
+
+section faulhaber
+
+/-- Faulhaber's theorem relating the sum of of p-th powers to the Bernoulli numbers.
+See https://proofwiki.org/wiki/Faulhaber%27s_Formula and [orosi2018faulhaber] for
+the proof provided here.-/
+theorem sum_range_pow (n p : ℕ) :
+    ∑ k in range n, (k : ℚ) ^ p
+  = ∑ i in range (p + 1), bernoulli i * (p + 1).choose i * n ^ (p + 1 - i) / (p + 1) :=
```

Comment:
We usually break equations like this ([the style guide](https://leanprover-community.github.io/contribute/style.html#structuring-definitions-and-theorems) says to put infix operators before a line break), though I think your way has its merits as well.
```suggestion
the proof provided here. -/
theorem sum_range_pow (n p : ℕ) :
  ∑ k in range n, (k : ℚ) ^ p =
    ∑ i in range (p + 1), bernoulli i * (p + 1).choose i * n ^ (p + 1 - i) / (p + 1) :=
```